packages/framework/src/Component/Elasticsearch/IndexRegistry.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Component\Elasticsearch;
+
+use Shopsys\FrameworkBundle\Component\Elasticsearch\Exception\ElasticsearchIndexException;
+
+class IndexRegistry
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Elasticsearch\AbstractIndex[]
+     */
+    protected $registeredIndexes;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Elasticsearch\AbstractIndex[] $indexes
+     */
+    public function __construct(array $indexes)
+    {
+        foreach ($indexes as $index) {
```

Comment:
You can use here a check of passed objects in array, like `Assert::allIsInstanceOf($indexes, AbstractIndex::class);`
to be more defensive.

What do you think?

Reply:
Interesting... Can find out if it is more paranoia or is the check really reasonable.

As this is by desing just kind of user input we could check it to stop it ASAP. If we are planning to autowire all children of `AbstractIndex` I would not check it here again.

Reply:
After discussion we decided to autowire Index classes automatically. 

This check is not necessary.